Library/Homebrew/test/language/java_spec.rb

```diff
@@ -1,7 +1,7 @@
 require "language/java"
 
 describe Language::Java do
-  describe "::java_home_env" do
+  describe "::java_home_env", :needs_macos do
```

Comment:
This should work on Linux, no?

Reply:
The current test which searches for the `version` string (e.g. `"1.8+"`) in `java_home` does not work. I can modify the test.

Reply:
Hmm. No, `:needs_macos` is more correct than `:needs_java`.
```ruby
java_home = described_class.java_home_env("blah")
```
On macOS this test is fine, but on Linux it fails unless Java version `blah` is installed, because on Linux it uses `JavaRequirement` to check that that particular version of Java is installed.

Reply:
@sjackman Fine with me as long as we have new tests for each of the generic cases that are implemented.

Reply:
I've added a test for the new `Language::Java.java_home` method.